### ISSUE TITLE
Incompatible Swift version - framework was built with 3.0.2 and the local version is 3.1.

### ISSUE THREAD

**Issue by spurdow** [SwiftyJSON/SwiftyJSON#830]
### What did you do?
updated to swift 3.1
remove `Carthage` folder
`carthage update --platform IOS`

### What did you expect to happen?

Should download and install dependency

### What actually happened instead?

```
*** Downloading SwiftyJSON.framework binary at "3.1.4"
*** Skipped installing SwiftyJSON.framework binary due to the error:
        "Incompatible Swift version - framework was built with 3.0.2 and the local version is 3.1."
*** Checking out SwiftyJSON at "3.1.4"
*** xcodebuild output can be found in /var/folders/1n/98nykp5d1597hwvr40kn93xw0000gn/T/carthage-xcodebuild.GjfyaR.log
```

### Environment

List the software versions you're using:

 - SwiftyJSON: *3.1.4*
 - Xcode Version: *8.3.1 (????)* (Open Xcode; In menubar: Xcode > About Xcode)
 - Swift Version: *3.1* (Open Xcode Preferences; Components > Toolchains. If none, use `Xcode Default`.)

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Carthage: *0.20.1* (Use `carthage version` in Terminal)

**Comment by miktav** [comment]
When running Carthage update, you need to specify an additional switch:

    carthage update --platform IOS --no-use-binaries

**Comment by pachun** [comment]
I think we can close this. It looks like this solution worked for someone else as well as myself. Additionally, it looks like 5 people upvoted the issue that haven't upvoted the solution, which means they may not know it exists. Check this out: [Tell Me When It Closes](https://tellmewhenitcloses.com/session/new).

**Comment by wongzigii** [comment]
@pachun @spurdow  Unfortunately, it is unsafe to publish a pre-built framework until Swift 4.0 ABI get stable. 
See https://github.com/Quick/Quick/issues/700#issuecomment-287955781

**Comment by spurdow** [comment]
In case anyone has the same problem, 
What i did was  executing this one `carthage update --platform iOS --no-use-binaries`

unfortunately it will not work yet most of the times. but after i removed Carthage folder and re run again `carthage update --platform iOS --no-use-binaries`. 
and also restart the xcode,  it should work. 

Maybe i was wrong because some dependencies I have also got error. So maybe executing `carthage update --platform iOS --no-use-binaries` should suffice. But if you have the same dependency list in your Cartfile then I think try what I did.

```
github "Alamofire/Alamofire" == 4.4.0
github "SwiftyJSON/SwiftyJSON" == 3.1.4
github "tristanhimmelman/AlamofireObjectMapper" == 4.0.2
github "Hearst-DD/ObjectMapper" == 2.2.3
github "realm/realm-cocoa" == 2.5.0 
github "kaandedeoglu/KDCircularProgress"
github "robbiehanson/CocoaAsyncSocket" "master"
github "mindsnacks/MSWeakTimer" "master"
github "emqtt/CocoaMQTT" "master"
github "cesarferreira/SwiftEventBus" == 2.2.0
github "dekatotoro/SlideMenuControllerSwift" "master"
github "kean/Nuke" ~> 5.0
```

**Comment by wongzigii** [comment]
Thanks for the following up! Glad you work fine. @spurdow

**Comment by agibsonuk** [comment]
I'm experiencing the same issue, however the `--no-use-binaries` update has no effect, nor does any sequence of restarts, clearing of derived data, or specifying `== 3.1.4` in the cartfile. I'm at a loss as to what to try next.

exact message is:
Module compiled with Swift 3.0.1 cannot be imported in Swift 3.1

Are there any other things I could try?

**Comment by spurdow** [comment]
@agibsonuk have you tried removing `Carthage` folder and re run again `carthage update --platform iOS --no-use-binaries`?

**Comment by agibsonuk** [comment]
Yes I had tried that with no affect, but I found the cause eventually. 

Using `xcrun swift -version` in terminal I identified that Xcode was using 3.0.2, which I think was linked to the fact I had an older version of Xcode Beta on my machine as well. This was causing SwiftyJSON to be built with 3.0.2 with `--no-use-binaries`

I removed the beta version which prompted Xcode to run Swift 3.1, so all good now.

Thanks!
